emukit/core/optimization/__init__.py

```diff
@@ -3,4 +3,6 @@
 
 
 from .acquisition_optimizer import AcquisitionOptimizer  # noqa: F401
```

Comment:
Maybe we should consider renaming AcquisitionOptimizer -> GradientAcquisitionOptimizer ?

Reply:
This is a better choice. Are you fine with such breaking changes?

Reply:
Yes, I think it should be fine

Reply:
It's renamed like suggested.